bcda/cclf/cclf.go

```diff
@@ -551,6 +563,18 @@ func validate(fileMetadata *cclfFileMetadata, cclfFileValidator map[string]cclfF
 	return nil
 }
 
+func validateFileName(fileName string) error {
+	filenameRegexp := regexp.MustCompile(`(T|P).*\.((?:A|T)\d{4})\.ACO.*\.ZC(0|8|9)Y(\d{2})\.(D\d{6}\.T\d{6})\d`)
+	filenameMatches := filenameRegexp.FindStringSubmatch(fileName)
```

Comment:
I think `MatchString()` might be a better fit, since we don't need to use the submatches for metadata. https://golang.org/pkg/regexp/#Regexp.MatchString